Move packages folder outside of src and next to bin.

chcosta opened the pull request.

joshfree commented:
why do we need to make this change?

chcosta commented:
This change will move the generated packages directory out of the "SourceDirectory" and into the "ProjectDirectory", so that is a sibling of src, bin, etc... This allows us to keep src in a pristine state with no generated files, which is cleaner and makes manual cleaning simpler.  Generated files (bin and packages) are easily found in the project directory.

krwq commented on src/System.Diagnostics.FileVersionInfo/tests/Assembly1.csproj:
```diff
@@ -30,7 +30,7 @@
   <ItemGroup>
     <Reference Include="System.Runtime">
-      <HintPath>..\..\packages\System.Runtime.4.0.20-beta-22412\lib\portable-wpa80+win80+net45+aspnetcore50\System.Runtime.dll</HintPath>
+      <HintPath>..\..\..\packages\System.Runtime.4.0.20-beta-22412\lib\portable-wpa80+win80+net45+aspnetcore50\System.Runtime.dll</HintPath>
```
Can we use $(PackagesDir) here?

mellinoe replied:
I believe this reference should be removed; this should be resolved from the packages.config file for this assembly. That is, unless it's not working for this assembly because it uses packages.Assembly1.config.

weshaggard replied:
If I remember correctly this project isn't building like the other projects so we may need to keep this reference (I'm happy to be wrong here). If we do keep it then please do update it to use $(PackagesDir) like @krwq suggests.

krwq commented:
Looks good :+1:
Will the package restore work both in VS and command line?
I was wondering if it makes sense to put them in the bin folder so we could easily manually clean by just deleting one folder.

weshaggard commented on src/System.Reflection.Metadata/src/System.Reflection.Metadata.csproj:
```diff
@@ -179,7 +179,7 @@
     <Reference Include="System.Collections.Immutable">
-      <HintPath>..\..\packages\System.Collections.Immutable.1.1.32-beta\lib\portable-net45+win8+wp8+wpa81\System.Collections.Immutable.dll</HintPath>
+      <HintPath>..\..\..\packages\System.Collections.Immutable.1.1.32-beta\lib\portable-net45+win8+wp8+wpa81\System.Collections.Immutable.dll</HintPath>
     </Reference>
```
Update this to use $(PackagesDir) as well.

weshaggard commented:
I think some further testing is needed to ensure scenarios like restoring packages from VS and such works. I suspect the package cache in VS lives next to the Solution file, or the global nuget cache. 

@terrajobst or @FiveTimesTheFun might know more about how this works as I couldn't find any clear reference in the nuget docs.

chcosta commented:
Updated PR based on feedback.

Added an entry to Nuget.Config to support package restores in VS.  While our build does not use "repositoryPath", Nuget packaging uses this key to support package restore.  It doesn't appear to have knowledge of our "PackagesDir" variable, so made it a relative path and validated functionality.

mmitche commented:
test this please (testing msbuild log gathering)

weshaggard commented:
LGTM thanks @chcosta
